packages/iov-socket/src/queueingstreamingsocket.ts

```diff
@@ -0,0 +1,32 @@
+// tslint:disable:readonly-array readonly-keyword
+import { StreamingSocket } from "./streamingsocket";
+
+/**
+ * A StreamingSocket that can queue requests.
+ */
+export class QueueingStreamingSocket extends StreamingSocket {
+  private queue: string[] = [];
+
+  public getQueueLength(): number {
+    return this.queue.length;
+  }
+
+  public async queueRequest(data: string): Promise<void> {
+    this.queue.push(data);
+    return this.processQueue();
+  }
+
+  public async processQueue(): Promise<void> {
+    let data: string | undefined = this.queue.shift();
+    while (data) {
```

Comment:
With `while ((data = this.queue.shift()) !== undefined)`, you can deduplicate the shifting

Reply:
True, but we have a lint rule against that because it's confusing to assign things in conditional statements: https://eslint.org/docs/rules/no-cond-assign

Reply:
I really don't like when tools prevent us from writing smart code. I checked the docs again and it turned out that eslint by default allows the version I suggested and just prevents = vs. == typos. The linter rule complaining here was tslint. See https://github.com/iov-one/iov-core/pull/1064